File: Sample/java/src/main/java/net/nend/sample/java/fullboard/ObservableWebView.java
package net.nend.sample.java.fullboard;

import android.content.Context;
import android.util.AttributeSet;
import android.webkit.WebView;

public class ObservableWebView extends WebView {

    private Callback mCallback;

    public ObservableWebView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    void setCallback(Callback callback) {
        mCallback = callback;
    }

    @Override
    protected void onScrollChanged(int l, int t, int oldl, int oldt) {
        int height = (int) Math.floor(getContentHeight() * getResources().getDisplayMetrics().density);
        int webViewHeight = getHeight();
        if (t > oldt && getScrollY() + webViewHeight >= height - 5) {
            mCallback.onScrollEnd();
        }
        super.onScrollChanged(l, t, oldl, oldt);
    }

    interface Callback {
        void onScrollEnd();
    }
}

File: Sample/kotlin/src/main/java/net/nend/sample/kotlin/fullboard/ObservableWebView.kt
package net.nend.sample.kotlin.fullboard

import android.content.Context
import android.util.AttributeSet
import android.webkit.WebView

class ObservableWebView(context: Context, attrs: AttributeSet) : WebView(context, attrs) {

    private var callback: Callback? = null

    internal fun setCallback(callback: Callback) {
        this.callback = callback
    }

    override fun onScrollChanged(l: Int, t: Int, oldl: Int, oldt: Int) {
        val height = Math.floor(
                (contentHeight * resources.displayMetrics.density).toDouble()).toInt()
        val webViewHeight = getHeight()
        callback?.let {
            if (t > oldt && scrollY + webViewHeight >= height - 5) {
                it.onScrollEnd()
            }
        }
        super.onScrollChanged(l, t, oldl, oldt)
    }

    internal interface Callback {
        fun onScrollEnd()
    }
}